src/backend/TrafficCourts/Staff.Service/Services/DisputeService.cs

```diff
@@ -263,4 +264,38 @@ public async Task RejectDisputeUpdateRequestAsync(long updateStatusId, Cancellat
         DisputantUpdateRequestRejected message = new(updateStatusId);
         await _bus.PublishWithLog(_logger, message, cancellationToken);
     }
+
+    /// <summary>
+    /// Returns a list of all disputes with pending update requests.
+    /// </summary>
+    /// <param name="cancellationToken"></param>
+    /// <returns></returns>
+    public async Task<ICollection<Dispute>> GetAllDisputesWithPendingUpdateRequestsAsync(CancellationToken cancellationToken)
+    {
+        ICollection<Dispute> disputes = new Collection<Dispute>();
+        ICollection<TrafficCourts.Common.OpenAPIs.OracleDataApi.v1_0.DisputantUpdateRequest> pendingDisputeUpdateRequests = await _oracleDataApi.GetDisputantUpdateRequestsAsync(null, Status.PENDING);
+
+        foreach (TrafficCourts.Common.OpenAPIs.OracleDataApi.v1_0.DisputantUpdateRequest disputantUpdateRequest in pendingDisputeUpdateRequests)
+        {
+            if (disputes.FirstOrDefault(x => x.DisputeId == disputantUpdateRequest.DisputeId) is null)
+            {
+                Dispute pendingDispute = await _oracleDataApi.GetDisputeAsync(disputantUpdateRequest.DisputeId);
+                disputes.Add(pendingDispute);
+            }
+        }
+
+        return disputes;
+
+    }
+
+    /// <summary>
+    /// Returns a list of all dispute update requests for a given dispute id.
+    /// </summary>
+    /// <param name="disputeId"></param>
+    /// <param name="cancellationToken"></param>
+    /// <returns></returns>
+    public async Task<ICollection<TrafficCourts.Common.OpenAPIs.OracleDataApi.v1_0.DisputantUpdateRequest>> GetDisputeUpdateRequestsAsync(long disputeId, CancellationToken cancellationToken)
+    {
+        return await _oracleDataApi.GetDisputantUpdateRequestsAsync(disputeId, null);

```

Comment:
I think you need to pass cancellationToken as well.

Reply:
done